api/base/filters.py

```diff
@@ -537,33 +346,43 @@ def param_queryset(self, query_params, default_queryset):
         """filters default queryset based on query parameters"""
         filters = self.parse_query_params(query_params)
         queryset = default_queryset
-
+        query_parts = []
+        sub_query_parts = []
         if filters:
             for key, field_names in filters.iteritems():
                 for field_name, data in field_names.iteritems():
                     operations = data if isinstance(data, list) else [data]
-                    for operation in operations:
-                        if isinstance(queryset, list):
+                    if isinstance(queryset, list):
+                        for operation in operations:
                             queryset = self.get_filtered_queryset(field_name, operation, queryset)
-                        else:
-                            queryset = self.filter_by_field(queryset, field_name, operation)
-        return queryset
+                    else:
+                        sub_query_parts.append(
+                            functools.reduce(operator.and_, [
+                                self.build_query_from_field(field_name, operation)
+                                for operation in operations
+                            ])
+                        )
+                try:
+                    sub_query = functools.reduce(operator.or_, sub_query_parts)
+                    query_parts.append(sub_query)
+                except TypeError:
```

Comment:
Really good point. I refactored this so it's more clear and no longer catching `TypeError`.